dev_guide/deployments.adoc

```diff
@@ -829,3 +952,283 @@ spec:
+described below:
+
+. Create the first shard of the application based on the example deployment images:
++
+----
+$ oc new-app openshift/deployment-example --name=ab-example-a --labels=ab-example=true SUBTITLE="shard A"
+----
+
+. Edit the newly created shard to set a label `ab-example=true` that will be
+common to all shards:
++
+----
+$ oc edit dc/ab-example-a
+----
++
+In the editor, add the line `ab-example: "true"` underneath `spec.selector` and
+`spec.template.metadata.labels` alongside the existing
+`deploymentconfig=ab-example-a` label. Save and exit the editor.
+
+. Trigger a re-deployment of the first shard to pick up the new labels:
++
+----
+$ oc deploy ab-example-a --latest
+----
+
+. Create a service that uses the common label:
++
+----
+$ oc expose dc/ab-example-a --name=ab-example --selector=ab-example=true
+----
++
+If you have the router installed, make the application available via a route (or
+use the service IP directly):
++
+----
+$ oc expose svc/ab-example
+----
++
+Browse to the application at `ab-example.<project>.<router_domain>` to verify
+you see the *v1* image.
+
+. Create a second shard based on the same source image as the first shard but
+different tagged version, and set a unique value:
++
+----
+$ oc new-app openshift/deployment-example:v2 --name=ab-example-b --labels=ab-example=true SUBTITLE="shard B" COLOR="red"
+----
+
+. Edit the newly created shard to set a label `ab-example=true` that will be
+common to all shards:
++
+----
+$ oc edit dc/ab-example-b
+----
++
+In the editor, add the line `ab-example: "true"` underneath `spec.selector` and
+`spec.template.metadata.labels` alongside the existing
+`deploymentconfig=ab-example-b` label. Save and exit the editor.
+
+. Trigger a re-deployment of the second shard to pick up the new labels:
++
+----
+$ oc deploy ab-example-b --latest
+----
+
+. At this point, both sets of pods are being served under the route. However,
+since both browsers (by leaving a connection open) and the router (by default,
+through a cookie) will attempt to preserve your connection to a back-end server,
+you may not see both shards being returned to you. To force your browser to one
+or the other shard, use the scale command:
++
+----
+$ oc scale dc/ab-example-a --replicas=0
+----
++
+Refreshing your browser should show *v2* and *shard B* (in red).
++
+----
+$ oc scale dc/ab-example-a --replicas=1; oc scale dc/ab-example-b --replicas=0
+----
++
+Refreshing your browser should show *v1* and *shard A* (in blue).
++
+If you trigger a deployment on either shard, only the pods in that shard will be
+affected. You can easily trigger a deployment by changing the `SUBTITLE`
+environment variable in either deployment config `oc edit dc/ab-example-a` or
+`oc edit dc/ab-example-b`. You can add additional shards by repeating steps 5-7.
++
+[NOTE]
+====
+These steps will be simplified in future versions of {product-title}.
+====
+
+[[shard-proxy-traffic-splitter]]
+=== Shard Proxy / Traffic Splitter
```

Comment:
We call it a "Shard Proxy" and a "proxy shard"? Same thing I assume, different way of referring to it?